### ISSUE TITLE
Product edit/view/admin very slow load

### ISSUE THREAD

**Issue by ricardo-lobo** [thirtybees/thirtybees#357]
The project that I am currently working on has a small issue when someone tries to edit/view a product on the Administration.

For reference, currently the following tables have this amount of rows:

feature: 4,838 rows
feature_product: 339,030 rows
feature_shop: 23,498 rows
feature_value: 41,721 rows

At the moment the shop has about 20,030 products. Everything works fine. The frontend is fast. The backoffice is also working fine, until you try to edit a product. Takes about 50 seconds to load a product. 

The server used is a XEON 8 Core, 32GB RAM, 480GB SSD.
I think the problem might be the query used to fetch all the features/values for the product.

I have a suggestion to make. In the product backoffice, instead of loading all the features, could we just load the active product features for the product? And if I wanted to add more features, maybe press a Button to load everything?

This would make life easier of someone just wants to edit price, or name. At the moment we only have 2 choices, wait for the product to full load everything, or disable the features tab on the controller.

**Comment by Traumflug** [comment]
As mentioned in the forum already, existing code looks much like earlier PrestaShop versions did actually load just one of the tabs, not all 15 of them. Each tab saves via its own Ajax call.

Which raises the question on why PS abandoned this. Doing the same error twice is rarely a good idea.

One obvious change is that with individual tabs, a user has to hit 'save' on each edited tab in case s/he edits more than one.

**Comment by ricardo-lobo** [comment]
Unfortunately my knowledge of the inner workings of Prestashop is still quite limited.

I for now, disabled the features tab and replaced it with a module that mimics the same behavior although with some important changes.

Instead of loading all 4800 features at once, it only loads the features that are being used by the product.

If I want to add another one, I have a search input that lets me search and add.

I'll try to improve it a bit more and then I'll share it here. Maybe it will serve as inspiration.

**Comment by Traumflug** [comment]
> Instead of loading all 4800 features at once, it only loads the features that are being used by the product.

I see. It's less about loading all the product related stuff, the issue is about loading stuff which is *not* related to the product (but might be added as part of the edit).

**Comment by ricardo-lobo** [comment]
> I see. It's less about loading all the product related stuff, the issue is about loading stuff which is not related to the product (but might be added as part of the edit).

Yes. Its really not efficient to load more than 45000 rows for a single product. This only tends to increase.

**Comment by firstred** [comment]
> Yes. Its really not efficient to load more than 45000 rows for a single product. This only tends to increase.

Hot damn! More than 45k :open_mouth: 

I know where it is coming from. Even though all the basic product info is loaded in one query, it is never joined on the right tables, so related objects often have to be loaded one by one, which causes the huge amount of queries.

I know this is going to take a while to fix, but I am assigning a high priority to this issue. Performance is important!

Thanks for sharing your stats, btw!
